src/services/creategeojsonbloodhound.js

```diff
@@ -58,7 +61,14 @@ ngeo.createGeoJSONBloodhound = function(options, opt_featureProjection,
       },
       filter: function(parsedResponse) {
         var featureCollection = /** @type {GeoJSONFeatureCollection} */
-            (parsedResponse);
+            (goog.object.clone(parsedResponse));
+
+        if (goog.isDef(opt_filter) && !goog.isNull(opt_filter)) {
+          var features = /* Array.<GeoJSONFeature> */
+              (featureCollection['features']);
```

Comment:
We have a type defined in an externs file for `GeoJSONFeatureCollection`, so you don't need to use the square bracket notation here.

Use this:

```js
  if (goog.isDef(opt_filter)) {
    var features = featureCollection.features.filter(opt_filter);
    featureCollection.features = features;
  }
```